src/shared/codelens/codeLensUtils.ts

```diff
@@ -26,6 +27,8 @@ import * as tsCodelens from './typescriptCodeLensProvider'
 
 export type Language = 'python' | 'javascript' | 'csharp' | 'java'
 
+export const STATE_NAME_SUPPRESS_CODELENSES = 'sam.suppressCodeLenses'
```

Comment:
"enable" is more common and thus more discoverable. "suppress" implies an active process.

```suggestion
export const STATE_NAME_ENABLE_CODELENSES = 'sam.enableCodeLenses'
```

Reply:
As stated above, I consider codelenses to be a default behavior and I'd rather have an explicit, conscious opt-out. Would `disable` work better?

Reply:
disable is definitely more common that suppress, and common names are a good default if there is no conscious reason to deviate

Reply:
The way I see it, this should only appear in your settings.json if you've taken some sort of action, and undefined === false. If you have `disabled: true`, that makes sense that you've disabled the value, but if you have `disabled: false` or no mention of `disabled` at all, that should imply that the feature wasn't disabled. As of now, I'm fairly sure an enabled checkbox (even by default) would write to the settings.json file?

Reply:
> this should only appear in your settings.json if you've taken some sort of action

Isn't that driven by the `default` field of configuration definition? Otherwise all non-boolean settings would write to settings.json

Reply:
OK did a little dig: we only have one precedent for something like this: `aws.stepfunctions.asl.format.enable`

That has a default value of `true` and does not show up in the global settings file when it starts out checked. I'll confirm that this is the case if I move my setting to enabled.